driver-sync/src/test/functional/com/mongodb/client/AbstractClientEncryptionRewrapManyDataKeyProseTest.java

```diff
@@ -147,4 +159,26 @@ public void rewrapWithSeparateClientEncryption(final String srcProvider, final S
         assertEquals(testString, clientEncryption1.decrypt(ciphertext));
         assertEquals(testString, clientEncryption2.decrypt(ciphertext));
     }
+
+    @Test
+    public void shouldThrowClientErrorWhenProviderIsNotSpecified() {
+        //given
+        String expectedErrorMessage = "Missing the provider but supplied a master key in the RewrapManyDataKeyOptions";
+        BsonDocument dstKey = MASTER_KEYS_BY_PROVIDER.get("aws");
+
+        ClientEncryption clientEncryption = getClientEncryption(ClientEncryptionSettings.builder()
+                .keyVaultMongoClientSettings(getMongoClientSettingsBuilder().build())
+                .keyVaultNamespace("keyvault.datakeys")
+                .kmsProviders(KMS_PROVIDERS)
+                .build());
+
+        RewrapManyDataKeyOptions rewrapManyDataKeyOptions = new RewrapManyDataKeyOptions().masterKey(dstKey);
+
+        //when
+        Executable executable = () -> clientEncryption.rewrapManyDataKey(new BsonDocument(), rewrapManyDataKeyOptions);
+
+        //then
+        MongoClientException mongoClientException = assertThrows(MongoClientException.class, executable);
+        assertEquals(expectedErrorMessage, mongoClientException.getMessage());
```

Comment:
From the usages I see, the assertion message is inlined. I think this is preferable because it limits the scope of the variable (to 0), and moves it closer to the usage.